### PR TITLE
feat: remove gitlab CI and create github CI

### PR DESCRIPTION
**Pull request by bolinocroustibat**
Related issue: https://github.com/etalab-ia/albert-tchapbot/issues/8

### CODE REVIEW

**Comment by bolinocroustibat** [comment]
There is still the .env file to create.
https://github.com/marketplace/actions/create-env-file

**Comment by bolinocroustibat** [comment]
@leoguillaume If you know GitHub Actions and secrets, let me know what you think, especially regarding the use of env files in GitHub (not sure on how we should achieve/convert that). Available to talk quickly about it after you have a quick look

**Comment by dtrckd** [comment]
@bolinocroustibat Can we set up the bot to always respond with the "maintenance message" in this PR, so we can prioritize merging it and validating the deployment pipeline (including any tests) before the other PRs are ready to be merged and tested?

**Comment by bolinocroustibat** [comment]
> @bolinocroustibat Can we set up the bot to always respond with the "maintenance message" in this PR, so we can prioritize merging it and validating the deployment pipeline (including any tests) before the other PRs are ready to be merged and tested?

Sure, but this is a draft PR and not ready to be merged, as I'm not sure on how to manage the env file in the CI/CD workflow file. I would need some help first on that

**Comment by bolinocroustibat** [comment]
PR is done and CI is working. We can fully switch to GitHub now.

File: .github/workflows/staging.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
+name: Deploy to staging when pushing on staging
+
+on:
+  push:
+    branches:
+      - staging
+      - feat/github-ci
```
Comment (dtrckd) [inline .github/workflows/staging.yml:7]: Why feat/github-ci ?

Comment (bolinocroustibat) [reply .github/workflows/staging.yml:7]: It was in order to test the branch. We can remove it later

File: .github/workflows/prod.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            mkdir -p /data/tchapbot
+            
+            export COMPOSE_FILE=/home/${{ secrets.CI_DEPLOY_USER }}/${{ env.CI_JOB_ID }}/docker-compose.yml
+            export COMPOSE_PROJECT_NAME=albert-tchapbot
+
+            docker compose down
+            docker tag albert/tchapbot:latest albert/tchapbot:old || true
+            docker image rm albert/tchapbot:latest || true
+            docker compose up --detach
+            docker image rm albert/tchapbot:old || true
```
Comment (dtrckd) [inline .github/workflows/prod.yml:76]: Surely something that I miss, but why do we rm the :latest while it has been renamed just before from :latest to :old ??

**Comment by dtrckd** [comment]
I am wondering, in fact the two pipelines are exactly the same except the env name.
Maybe this CI can be simplified by having just one single file if there is a way to select the environement as the name of the branch. It should be easy. What do you think about it ? @bolinocroustibat

**Comment by bolinocroustibat** [comment]
> I am wondering, in fact the two pipelines are exactly the same except the env name.
> Maybe this CI can be simplified by having just one single file if there is a way to select the environement as the name of the branch. It should be easy. What do you think about it ? @bolinocroustibat 

Yes it should be doable indeed, choosing the environment depending on the branch. I'll have a look on that tomorrow

File: .github/workflows/prod.yml (hunk 2 above)
Comment (bolinocroustibat) [reply .github/workflows/prod.yml]: I took the logic from the previous CI (.gitlab_ci.yml) which was working fine and I didn't want to change the container cache logic at all. For this question you should ask @leoguillaume

**Comment by bolinocroustibat** [comment]
> > I am wondering, in fact the two pipelines are exactly the same except the env name.
> > Maybe this CI can be simplified by having just one single file if there is a way to select the environement as the name of the branch. It should be easy. What do you think about it ? @bolinocroustibat
> 
> Yes it should be doable indeed, choosing the environment depending on the branch. I'll have a look at that tomorrow

@dtrckd It's done on the same PR. I realized there were a few other stuff not working. Have a look when you have the time.

**Review by dtrckd — APPROVED** [review]
The build_and_deploy job, in a single file, looks really cleaner and simpler now, good job!

**Pull request merged**